File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ArrayList<Integer> ans = new ArrayList<>();
        ListNode temp = head;
        while(temp!=null){
            ans.add(temp.val);
            temp=temp.next;
        }
        int target = ans.size()-n;

        ListNode dummy = new ListNode(0);
        ListNode dummy1 = dummy;
        for(int i=0; i<ans.size(); i++){
            int v = ans.get(i);
           if(i==target) continue;


            else{
                dummy1.next = new ListNode(v);
                dummy1=dummy1.next;
            }
        }
        return dummy.next;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: Optional[ListNode]
        :type n: int
        :rtype: Optional[ListNode]
        """

        count=0
        temp = head
        while(temp!=None):
            temp = temp.next
            count+=1
        target = count-n
        ans = []
        t = head
        while t!=None:
            ans.append(t.val)
            t=t.next
        answer = []
        for i in range(len(ans)):
            if target == i:
                continue
            else:
                answer.append(ans[i])
        dummy = ListNode(0)
        dummy1 = dummy
        for var in answer:
            dummy1.next = ListNode(var)
            dummy1 = dummy1.next
        return dummy.next

        